appengine/cloudsql/createTables.js

```diff
@@ -13,71 +13,127 @@
+const process = require('process');
+
 // [START createTables]
 // [START setup]
-const mysql = require('mysql');
+const Knex = require('knex');
 const prompt = require('prompt');
 // [END setup]
 
 // [START createTable]
-const SQL_STRING = `CREATE TABLE visits (
-  id INT UNSIGNED NOT NULL AUTO_INCREMENT,
-  timestamp DATETIME NULL,
-  userIp VARCHAR(46) NULL,
-  PRIMARY KEY (id)
-);`;
-
 /**
  * Create the "visits" table.
  *
- * @param {object} connection A mysql connection object.
- * @param {function} callback The callback function.
+ * @param {object} knex A Knex client object.
  */
-function createTable (connection, callback) {
-  connection.query(SQL_STRING, callback);
+function createTable (knex) {
+  return knex.schema.createTable('visits', (table) => {
+    table.increments();
+    table.timestamp('timestamp');
+    table.string('userIp');
+  })
+  .then(() => {
+    console.log(`Successfully created 'visits' table.`);
+    return knex;
+  })
+  .catch((err) => {
+    console.error(`Failed to create 'visits' table:`, err);
+    return knex;
+  });
 }
 // [END createTable]
 
-// [START getConnection]
-const FIELDS = ['user', 'password', 'database'];
-
+// [START postgresGetConnection]
 /**
  * Ask the user for connection configuration and create a new connection.
- *
- * @param {function} callback The callback function.
  */
-function getConnection (callback) {
-  prompt.start();
-  prompt.get(FIELDS, (err, config) => {
-    if (err) {
-      callback(err);
-      return;
-    }
+function postgresGetConnection () {
+  const FIELDS = ['user', 'password', 'database'];
+  return new Promise((resolve, reject) => {
+    prompt.start();
+    prompt.get(FIELDS, (err, config) => {
+      if (err) {
+        reject(err);
+      }
 
-    const user = encodeURIComponent(config.user);
-    const password = encodeURIComponent(config.password);
-    const database = encodeURIComponent(config.database);
-
-    const uri = `mysql://${user}:${password}@127.0.0.1:3306/${database}`;
-    callback(null, mysql.createConnection(uri));
+      // Connect to the database
+      resolve(Knex({
+        client: 'pg',
+        connection: config
+      }));
+    });
   });
 }
-// [END getConnection]
+// [END postgresGetConnection]
+
+// [START mysqlGetConnection]
+/**
+ * Ask the user for connection configuration and create a new connection.
+ */
+function mysqlGetConnection () {
+  const FIELDS = ['user', 'password', 'database'];
+  return new Promise((resolve, reject) => {
+    prompt.start();
+    prompt.get(FIELDS, (err, config) => {
+      if (err) {
+        return reject(err);
+      }
 
-// [START main]
-getConnection((err, connection) => {
-  if (err) {
-    console.error(err);
-    return;
-  }
-  createTable(connection, (err, result) => {
-    connection.end();
-    if (err) {
-      console.error(err);
-      return;
-    }
-    console.log(result);
+      // Connect to the database
+      return resolve(Knex({
+        client: 'mysql',
```

Comment:
Again, I think we only need one connection method, just set the `client` Knex option to `process.env.SQL_CLIENT`.